src/middleware/api.js

```diff
@@ -0,0 +1,64 @@
+import { serverBaseUrl } from '../config';
+import { getToken } from '../utils/tokenStorage';
+
+const STATUS_CODES = {
+  UNAUTHORIZED: 401,
+};
+
+export default () => dispatch => async action => {
+  const { endpoint, method, types, content } = action;
+  if (typeof types === 'undefined') {
+    // So the middleware doesn't get applied to every single action
+    return dispatch(action);
+  }
+
+  const requestType = types.request;
```

Comment:
nitpick:
maybe a destructure call is easier to read here

Reply:
good suggestion, adding it